File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.BandedGrid.ViewInfo;
using DevExpress.Utils.Drawing;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
                private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i) });
            return tbl;
        }
        

        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
            gridControl1.PaintEx += GridControl1_PaintEx;
        }

        private void GridControl1_PaintEx(object sender, DevExpress.XtraGrid.PaintExEventArgs e)
        {
            foreach (GridBand band in advBandedGridView1.Bands)
            {
                DrawObjectBorder(e.Cache, GetBandBounds(band));
            }
            foreach (GridColumn col in advBandedGridView1.VisibleColumns)
            {
                DrawObjectBorder(e.Cache, GetColumnBounds(col));
            }
        }
        
        private void DrawObjectBorder(GraphicsCache g, Rectangle bounds)
        {
            g.DrawRectangle(new Pen(Brushes.Black, 2), bounds);
        }

        public BandedGridViewInfo ViewInfo
        {
            get
            {
                return advBandedGridView1.GetViewInfo() as BandedGridViewInfo;
            }
        }

        private Rectangle GetBandBounds(GridBand band)
        {
            HeaderObjectInfoArgs info = ViewInfo.BandsInfo[band];
            return info == null? Rectangle.Empty : info.Bounds;
        }
        private Rectangle GetColumnBounds(GridColumn col)
        {
            HeaderObjectInfoArgs info = ViewInfo.ColumnsInfo[col];
            return info == null ? Rectangle.Empty : info.Bounds;
        }

    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.BandedGrid
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.BandedGrid.ViewInfo
Imports DevExpress.Utils.Drawing

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form

				Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i) })
			Next i
			Return tbl
				End Function


		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = CreateTable(20)
			AddHandler gridControl1.PaintEx, AddressOf GridControl1_PaintEx
		End Sub

		Private Sub GridControl1_PaintEx(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.PaintExEventArgs)
			For Each band As GridBand In advBandedGridView1.Bands
				DrawObjectBorder(e.Cache, GetBandBounds(band))
			Next band
			For Each col As GridColumn In advBandedGridView1.VisibleColumns
				DrawObjectBorder(e.Cache, GetColumnBounds(col))
			Next col
		End Sub

		Private Sub DrawObjectBorder(ByVal g As GraphicsCache, ByVal bounds As Rectangle)
			g.DrawRectangle(New Pen(Brushes.Black, 2), bounds)
		End Sub

		Public ReadOnly Property ViewInfo() As BandedGridViewInfo
			Get
				Return TryCast(advBandedGridView1.GetViewInfo(), BandedGridViewInfo)
			End Get
		End Property

		Private Function GetBandBounds(ByVal band As GridBand) As Rectangle
			Dim info As HeaderObjectInfoArgs = ViewInfo.BandsInfo(band)
			Return If(info Is Nothing, Rectangle.Empty, info.Bounds)
		End Function
		Private Function GetColumnBounds(ByVal col As GridColumn) As Rectangle
			Dim info As HeaderObjectInfoArgs = ViewInfo.ColumnsInfo(col)
			Return If(info Is Nothing, Rectangle.Empty, info.Bounds)
		End Function

	End Class
End Namespace